x/profiles/module_ibc.go

```diff
@@ -114,32 +154,82 @@ func (am AppModule) OnChanCloseConfirm(
 	return nil
 }
 
+// -------------------------------------------------------------------------------------------------------------------
+
 // OnRecvPacket implements the IBCModule interface
 func (am AppModule) OnRecvPacket(
 	ctx sdk.Context,
-	modulePacket channeltypes.Packet,
+	packet channeltypes.Packet,
 ) (*sdk.Result, []byte, error) {
-	var packetData types.LinkChainAccountPacketData
-	if err := types.ProtoCdc.UnmarshalJSON(modulePacket.GetData(), &packetData); err != nil {
-		return nil, nil, sdkerrors.Wrapf(sdkerrors.ErrUnknownRequest, "cannot unmarshal packet data: %s", err.Error())
+	var ack channeltypes.Acknowledgement
+	var err error
+
+	// Try handling the chain link packet data
+	ack, err = am.HandlePacket(ctx, packet, handleOracleRequestPacketData, handleLinkChainAccountPacketData)
```

Comment:
@dadamu I'm going to merge the PR, but just as a reference I've changed how multiple packet data types are handled. You can see the new logic here